plugins/catalog-backend-module-bitbucket-server/src/module/catalogModuleBitbucketServerEntityProvider.test.ts

```diff
@@ -39,12 +32,12 @@ describe('catalogModuleBitbucketServerEntityProvider', () => {
       },
     };
     const runner = jest.fn();
-    const scheduler = {
-      createScheduledTaskRunner: (schedule: TaskScheduleDefinition) => {
+    const scheduler = mockServices.scheduler.mock({
+      createScheduledTaskRunner(schedule) {
```

Comment:
Hm. Bit of a matter of taste, but would it make sense to have a default impl for this method, not just `jest.fn()` but `jest.fn(() => ({ run: jest.fn() }))`? Maybe it's hard to always do that extra step, depending on the complexity of the interfaces at hand. And I guess that that implementation then becomes part of a "promised contract" and we'll break shit if we slightly change that later on when tests rely on it doing a certain thing. Maybe it's best as-is, forcing people to ALWAYS explicitly define the implementation, visibly in the test.

I just thought it'd make the mock slightly more directly usable, without writing a fake implementation at all here. Then you could also assert directly on the mock interface of `scheduler.createScheduledTaskRunner` below if you wanted to (doesn't necessarily make a prettier test here though ...).

Reply:
Yep also thought about providing some sane defaults for the config service, like `app.baseUrl` etc. At least at this point I find it much safer to not do that though. We'd ofc break a lot of existing tests of we want to roll something like this back for some reason, and I do think there's a large risk of confusion here. There's no good way to be informed what methods provide default mock implementations and which don't. Overall I feel it's safer to consistently never provide any default implementation.

Reply:
Yeah that's probably prudent